### PR TITLE
initial commit of new quickstart

### PR DESCRIPTION
**Pull request by Pouriya-Inmanta**
# Description

Replaces Drupal model of quickstart with SR Linux.

### CODE REVIEW

**Comment by bartv** [comment]
@Pouriya-Inmanta I would also like to add a daily test like we do with the current quickstart:

https://jenkins.inmanta.com/job/integration-tests/job/quickstart-docker
https://github.com/inmanta/quickstart-docker/blob/master/ci/Jenkinsfile

Run the full main.cf and then do checks using ssh or the json api to see if the config / operational state of the switches is correct.

File: docs/quickstart.rst [hunk 1]
```diff
@@ -3,361 +3,556 @@
 Quickstart
 ***************
 
-This tutorial gets you started with the Inmanta orchestration tool.
-
 Inmanta is intended to manage complex infrastructures, often in the cloud or other virtualized environments.
-In this guide, we go for a less complex setup: install the Drupal CMS on two VM-like containers.
-First, we use Docker to set up a basic environment with two empty VM-like containers, an Inmanta server and a postgres server used by inmanta as a database.
-Then, we use Inmanta to install Drupal on these VM-like containers.
+In this guide, we go for a less complex setup: Installing `containerlab <https://containerlab.dev/>`_ and configuring `SR Linux <https://learn.srlinux.dev/>`_ containers using **Inmanta service orchestrator** and ``gNMI``.
```
Comment (bartv) [inline docs/quickstart.rst:7]:
```suggestion
In this guide we start simple and manage a 3 node CLOS network with spine and two leaf switches. In this guide we will install `containerlab <https://containerlab.dev/>`_ and configure `SR Linux <https://learn.srlinux.dev/>`_ containers using **Inmanta open source orchestrator** and ``gNMI``.
```

File: docs/quickstart.rst [hunk 2]
```diff
@@ -3,361 +3,556 @@
...
+        name: SR Linux Examples
+        description: Provides examples for the SR Linux module
+        author: Inmanta
+        author_email: code@inmanta.com
+        license: ASL 2.0
+        copyright: 2022 Inmanta
+        modulepath: libs
+        downloadpath: libs
+        repo:
+            - https://github.com/inmanta/
```
Comment (bartv) [inline docs/quickstart.rst:55]: TODO: we need to fix this to provide a repo that contains the srlinux and yang modules.

File: docs/quickstart.rst [hunk 3]
```diff
@@ -3,361 +3,556 @@
...
+
+Depending on your system's horsepower, give them a few seconds/minutes to fully boot-up.
+
+
+Connecting to the containers
+______________________________
+
+At this stage, you should be able to view the **Web Console** by navigating to:
+
+http://172.30.0.3:8888/dashboard
```
Comment (bartv) [inline docs/quickstart.rst:106]:
```suggestion
http://172.30.0.3:8888/console
```

File: docs/quickstart.rst [hunk 4]
```diff
@@ -3,361 +3,556 @@
...
+        # Create a project called test
+        inmanta-cli --host 172.30.0.3 project create -n test
+        # Create an environment called SR_Linux
+        inmanta-cli --host 172.30.0.3 environment create -p test -n SR_Linux --save
+
+
+The first option, ``inmanta-cli``, will automatically create a ``.inmanta`` file that contains the required information about the server and environment ID. The compiler uses this file to find the server and to export to the right environment.
+
+
+2. Using the Web Console: Connect to the Inmanta container http://172.30.0.3:8888/dashboard, click on the `Create new environment` button, provide a name for the project and the environment then click `submit`.
```
Comment (bartv) [inline docs/quickstart.rst:191]:
```suggestion
2. Using the Web Console: Connect to the Inmanta container http://172.30.0.3:8888/console, click on the `Create new environment` button, provide a name for the project and the environment then click `submit`.
```

**Review by bartv — CHANGES_REQUESTED** [review]

File: docs/quickstart.rst (hunk 1 above)
Comment (Pouriya-Inmanta) [reply docs/quickstart.rst:7]: `In this guide` is repeated here. it kind of sounds unnatural.  
Let's apply the suggestion here and then I'll rectify it later when everything is in order.

Comment (bartv) [reply docs/quickstart.rst]: yeah, the two times in this guide is a mistake.

File: docs/quickstart.rst [hunk 5]
```diff
@@ -3,361 +3,556 @@
...
+____________________________________
+
+To deploy the project, we must first register it with the management server by creating a project and an environment. We have covered this earlier at `Create an Inmanta project and an environment`_ section.
+
+Export the ``interafces`` configuration model to the Inmanta server:
 
 .. code-block:: sh
 
-    inmanta project install
+    inmanta -vvv export interfaces.cf
```
[the remark this answers is not in the document]
Comment (Pouriya-Inmanta) [reply docs/quickstart.rst:417]: good catch. Fixing it.

[the remark this answers is not in the document]
Comment (Pouriya-Inmanta) [reply docs/quickstart.rst]: fixed in https://github.com/inmanta/inmanta-core/pull/4333/commits/d7b8d8ec60e77f6b8114b60dd54b6d906534af6a

**Comment by bartv** [comment]
Blocked by https://github.com/inmanta/irt/issues/1190

**Review by bartv — CHANGES_REQUESTED** [review]
The deploy fails for me. Still figuring out why.

File: docs/quickstart.rst [hunk 6]
```diff
@@ -3,361 +3,556 @@
...
+1. `Install Docker <https://docs.docker.com/install/>`_.
+2. `Install Containerlab <https://containerlab.dev/install/>`_.
+3. Prepare a development environment by creating a `python virtual environment` and installing Inmanta:
+
+   .. code-block:: sh
+
+       mkdir -p ~/.virtualenvs
+       python3 -m venv ~/.virtualenvs/srlinux
+       source ~/.virtualenvs/srlinux/bin/activate
+       pip install inmanta-core
```
[the remark this answers is not in the document]
Comment (Pouriya-Inmanta) [reply docs/quickstart.rst:33]: I believe we got the reason for installing `inmanta-core` on Slack yesterday?

File: docs/quickstart.rst [hunk 7]
```diff
@@ -3,361 +3,556 @@
...
+
+``Docker``, ``Containerlab`` and ``Inmanta`` need to be installed on your machine and our ``SR Linux`` repository has to be cloned in order to proceed. Please make sure to follow the links below to that end.
+
+1. `Install Docker <https://docs.docker.com/install/>`_.
+2. `Install Containerlab <https://containerlab.dev/install/>`_.
+3. Prepare a development environment by creating a `python virtual environment` and installing Inmanta:
+
+   .. code-block:: sh
+
+       mkdir -p ~/.virtualenvs
```
[the remark this answers is not in the document]
Comment (Pouriya-Inmanta) [reply docs/quickstart.rst:30]: Do you mean installing in that folder is complex?  
I am following the same format as per the training and our videos.

File: docs/quickstart.rst [hunk 8]
```diff
@@ -3,361 +3,556 @@
...
+Setting up the LAB
+_________________________
+
+Go to the `SR Linux` folder and then `containerlab` to spin-up the containers:
 
 .. code-block:: sh
 
-    git clone https://github.com/inmanta/quickstart-docker.git
-    cd quickstart-docker
+    cd Networking/SR\ Linux/containerlab
```
[the remark this answers is not in the document]
Comment (Pouriya-Inmanta) [reply docs/quickstart.rst]: fixed in https://github.com/inmanta/inmanta-core/pull/4333/commits/44087c5057d06b4a27725222896be836f8edce50

File: docs/quickstart.rst [hunk 9]
```diff
@@ -3,361 +3,556 @@
...
+       pip install inmanta-core
+
+4. Clone the `SR Linux examples <https://github.com/inmanta/examples/tree/master/Networking/SR%20Linux>`_ repository:
+
+   .. code-block:: sh
+
+       git clone https://github.com/inmanta/examples.git
+
+
+The cloned repository contains a **project.yml**, which looks like this:
```
[the remark this answers is not in the document]
Comment (Pouriya-Inmanta) [reply docs/quickstart.rst]: fixed in https://github.com/inmanta/inmanta-core/pull/4333/commits/44087c5057d06b4a27725222896be836f8edce50

File: docs/quickstart.rst [hunk 10]
```diff
@@ -3,361 +3,563 @@
...
+.. code-block::
+
+    Welcome to the srlinux CLI.
+    Type 'help' (and press <ENTER>) if you need any help using this.
 
-    docker-compose will lock the current terminal and use it for output from all 4 containers.
-    You will need to open a new terminal to continue with this quickstart
 
-To get an interactive shell on the Inmanta server (this will be needed later):
+    --{ running }--[  ]--
```
Comment (DariushMoh) [inline docs/quickstart.rst:157]: When trying the command in spine, I do not observe this output. Instead I get "--{ candidate shared default }--[  ]--" (without quotes)

Comment (Pouriya-Inmanta) [reply docs/quickstart.rst:157]: Could you verify from the console that the agent actually pushed the config to the spine?

Comment (edvgui) [reply docs/quickstart.rst:157]: But that this before the orchestrator actually does anything, isn't it?
 The issue reported here is I think that when you type `enter candidate` the cli doesn't show `--{ running }--[  ]--` but `--{ candidate shared default }--[ ]--` (which seems correct to me)

Comment (Pouriya-Inmanta) [reply docs/quickstart.rst:157]: I see, good catch!

**Review by bartv — APPROVED** [review]

**Review by edvgui — CHANGES_REQUESTED** [review]
It looks like a merge should be done, the diff contains way to many things

**Review by edvgui — APPROVED** [review]

**Pull request closed without merging**